Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -1008,3 +1023,12 @@ internal struct RedirectHandler<ResponseType> {
         }
     }
 }
+
+class IdlePoolConnectionHandler: ChannelInboundHandler, RemovableChannelHandler {
+    typealias InboundIn = NIOAny
+    func userInboundEventTriggered(context: ChannelHandlerContext, event: Any) {
+        if let idleEvent = event as? IdleStateHandler.IdleStateEvent, idleEvent == .write {
+            context.close(promise: nil)
+        }
```

Comment:
you need to forward all other events here in an `else` branch

Reply:
Thank you